lib/sycamore/sycamore/writer.py

```diff
@@ -367,17 +372,38 @@ def duck(self, db_name=None, table_name="aryn_table", **kwargs):
                     .sketch(window=17)
                 )
 
-                ds.write.duckdb(url="duckdb://")
+            ds.write.duckdb(table_name="duckdb")
         """
         from sycamore.writers.duckdb_writer import DuckDBDocumentWriter, DuckDBClientParams, DuckDBTargetParams
+        import duckdb
 
-        client_params = DuckDBClientParams(db_name=db_name)
-        target_params = DuckDBTargetParams(table_name=table_name)
-
+        csv_location = csv_directory_location if csv_directory_location is not None else "tmp/duckdb"
+        table_name = table_name if not None else "data"
+        client_params = DuckDBClientParams()
+        target_params = DuckDBTargetParams(parquet_location=csv_location)
         ddb = DuckDBDocumentWriter(
             self.plan, client_params=client_params, target_params=target_params, name="duck_write_documents", **kwargs
         )
-        ddb.execute()
+        ddb.execute().materialize()
```

Comment:
add a `execute=False` option, as discussed. This may require turning the CSV import thingy into a transform of its own or something